code/modules/projectiles/ammunition/boxes.dm

```diff
@@ -18,14 +18,16 @@
 
 /obj/item/ammo_magazine/speedloader/on_update_icon()
 	. = ..()
-	if(!length(stored_ammo))
+	var/ammo_count = get_stored_ammo_count()
+	if(!ammo_count)
 		return
+	create_initial_contents() // Not ideal, but we need instances for the icon gen.
```

Comment:
you could set lazyload_contents = FALSE on speedloaders instead? theoretically it's possible to do this without it because all you need is `marking_color`, `bullet_color`, `color`, etc. but like... it's just speedloaders, it isn't a big deal